### PR TITLE
Implements xADGroup

### PR DESCRIPTION
**Pull request by iainbrighton**
* Implements new xADGroup resource to fix issue #33.
 * Adds test coverage for xADGroup
 * Updates readme with new functionality
* As per the discussions, this only creates AD groups
 * The xADGroupMember functionality will be merged in afterwards

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -9,6 +9,8 @@ install:
   - cd DscResource.Tests
   - ps: Import-Module .\TestHelper.psm1 -force
   - ps: Pop-Location
+  - ps: |
```
Comment (vors) [inline appveyor.yml]: nitpick: keep it in one line: `- ps: Add-WindowsFeature RSAT-AD-PowerShell`

**Comment by vors** [comment]
There is `Error initializing default drive: 'Unable to find a default server with Active Directory Web Services running.'` warning message in AppVeyor build https://ci.appveyor.com/project/PowerShell/xactivedirectory/build/2.7.83.0 
@iainbrighton can we safely ignore it in test runs because everything is mocked? Do we need `Add-WindowsFeature RSAT-AD-PowerShell` in this case?

**Comment by vors** [comment]
LGTM

**Comment by iainbrighton** [comment]
@vors You'll get that error as the build server is not domain joined and therefore cannot mount the default drive. I'll see if there's a way of turning that warning off.

As to whether we need to add the RSAT tools - yes. Without it we can't mock the 'Microsoft.ActiveDirectory.Management.ADIdentityNotFoundException' type that used to detect whether a group exists or not. We could catch all exceptions and assume it's an AD object not found exception, but that would mask any other errors.

**Comment by vors** [comment]
Got it, thank you for explanation! I'm going ahead with merging it.

**Pull request merged**